documentation/versioning.md

```diff
@@ -0,0 +1,116 @@
+
+## What Is aa Azure Service?
+
+An Azure service is a set of operations that version uniformly (in perpetuity).  
+- A customer client app is expected to use only one version of a service at any given point in time.  
+- This means that any public or private preview service version must be a superset of that last stable service version (unless a preview version introduces a breaking change).  
+- Services can version & be retired independently from each other. This allows services to be agile and removes coordination between sub-teams.  
+- Customers consciously select and target specific service versions.
+    - This is important for the public cloud customers as all datacenters may not have the same service versions deployed. 
+    - It is also important for Azure Stack, government cloud, and other air-gapped/non-public cloud customers which also may not have the same service versions deployed.  
+
+## An Azure Resource Provider and its 1+ Azure Services
+
+A single Azure Resource Provider (RP) has an RP namespace. This RP namespace can contain 1+ control-plane services and 0+ data-plane service; each service versions as described above . Today, we want the GitHub specs repo to be structured as follows for control-plane & data-plane contracts:
+```
+specification/
+└── <orgName1>/
+    ├── cspell.yaml
+    └── resource-manager/
+        ├── readme.md			ß NOTE: For ARM schema validation; see bullet #2 below
+        └── <RPNamespace>/		ß NOTE: Control-plane only (not data-plane) 
+            └── <ServiceName1>/	ß CX-facing service name; each version gets Docs & SDK package
+                ├── tspconfig.yaml	ß TypeSpec files go here (examples below)
+                ├── main.tsp
+                ├── models.tsp
+                ├── readme.md		ß autorest readme with YAML blocks
+                └── examples/		ß TypeSpec example folder
+                    └── <api-version>/	ß One folder per service version described in TypeSpec
+                        └── <example .json files> 
+                └── preview/ and stable/
+                    └── <api-version>/	ß One folder per service version described in Swagger. These folders are created and populated by compiling the TypeSpec folder for the service.
+                        ├── <swagger .json files>
+                        └── examples/	ß Swagger example folder
+                            └── <example .json files>
+            └── <ServiceName2>/	// CX-facing service name; contents identical to above structure
+    └── data-plane/
+        └── <ServiceName3>/	// CX-facing service name; contents identical to above structure
+```
+In this structure, Azure services are presented as an RPNamespace that is used to manage services (both with customer-facing names)., each service has versions (some preview and some stable), and each version has its own API contract, documentation, and (beta/GA) SDK package(s).   
```

Comment:
Service versions are preview or GA, SDK packages are beta or stable.

Reply:
Thanks for the suggestion, I have changed everywhere for stable to GA when it's about service version.